### PR TITLE
IOHKLE-179: Support build variants

### PR DESCRIPTION
**Pull request by DCRichards**
* Support for `test` build variant in makefile, for now we'll assume anything else is production, but this can be easily changed.
* Usage: `make load VARIANT=test` to specify a test build, otherwise production.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,14 +37,20 @@ all: default
...
+
 DEFINES   += OS_IO_SEPROXYHAL IO_SEPROXYHAL_BUFFER_SIZE_B=128
 DEFINES   += HAVE_BAGL HAVE_PRINTF
-DEFINES   += INS_SIGN_TX_FUNC INS_SET_TX_FUNC
-DEFINES   += INS_GET_PUBLIC_KEY_FUNC
-#DEFINES   += INS_GET_WALLET_INDEX_FUNC
-#DEFINES   += INS_GET_RND_PUB_KEY_FUNC
-#DEFINES   += INS_HASH_FUNC
-#DEFINES   += INS_CBOR_DECODE_TEST_FUNC
+DEFINES   += INS_HASH_FUNC
```
Comment (StaceC) [inline Makefile]: Move INS_HASH_FUNC into the test functions. I'll rename it at some point to INS_HASH_TEST_FUNC

Comment (DCRichards) [reply Makefile]: Boom, will do!

File: Makefile [hunk 2]
```diff
@@ -37,14 +37,20 @@ all: default
...
+  DEFINES   += INS_GET_PUBLIC_KEY_FUNC
+  DEFINES   += INS_GET_WALLET_INDEX_FUNC
+  DEFINES   += INS_GET_RND_PUB_KEY_FUNC
+endif
+
 DEFINES   += OS_IO_SEPROXYHAL IO_SEPROXYHAL_BUFFER_SIZE_B=128
 DEFINES   += HAVE_BAGL HAVE_PRINTF
-DEFINES   += INS_SIGN_TX_FUNC INS_SET_TX_FUNC
-DEFINES   += INS_GET_PUBLIC_KEY_FUNC
-#DEFINES   += INS_GET_WALLET_INDEX_FUNC
```
Comment (StaceC) [inline Makefile:44]: Probably want to keep the funcs that have been commented out still commented out on the change, as it means they will not fit in a single build currently. Unless you've tested?

**Review by StaceC — APPROVED** [review]
That's awesome Dave

File: Makefile (hunk 2 above)
Comment (DCRichards) [reply Makefile:44]: Done!

**Comment by DCRichards** [comment]
Declined due to change in direction with this

**Pull request closed without merging**